gtk_printer_get_backend: assertion 'GTK_IS_PRINTER (printer)' failed

mcatanzaro opened the pull request:
#### 732b9298ef904661970fe96ff5e45fe044ea82ce
<pre>
gtk_printer_get_backend: assertion &apos;GTK_IS_PRINTER (printer)&apos; failed
<a href="https://bugs.webkit.org/show_bug.cgi?id=296238">https://bugs.webkit.org/show_bug.cgi?id=296238</a>

Reviewed by Carlos Garcia Campos.

The GtkPrintBackend is now bundled into libgtk rather than installed
separately on the filesystem, and its name has changed.
WebKitPrintOperation needs to be updated accordingly, or printing does
not work.

Also, bail out if filePrinter is nullptr.

I tested this by manually hacking shouldUsePortal to return true.

Canonical link: <a href="https://commits.webkit.org/297692@main">https://commits.webkit.org/297692@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/72751a3026f3c1161e12b0ccb0744d7a3b965670

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/112570 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/32302 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/22780 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/118769 "Built successfully") | [  ~~🛠 win~~](https://ews-build.webkit.org/#/builders/59/builds/63029 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/32954 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/40865 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/85680 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/59/builds/63029 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/115517 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/26297 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/101272 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/65982 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/25596 "Passed tests") | | [  ~~🛠 wpe-cairo~~](https://ews-build.webkit.org/#/builders/65/builds/62528 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/95689 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/136/builds/19478 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/121990 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/39644 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/29533 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/94546 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/40026 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/97504 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/94283 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/24069 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/39398 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/17202 "Passed tests") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/35732 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/39532 "Built successfully") | [  ~~🛠 mac-safer-cpp~~](https://ews-build.webkit.org/#/builders/120/builds/45020 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/39169 "Built successfully") | | | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/129/builds/42504 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/40910 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

carlosgcampos commented on Source/WebKit/UIProcess/API/gtk/WebKitPrintOperation.cpp:
```diff
@@ -430,7 +430,14 @@ static GRefPtr<GtkPrinter> findFilePrinter(GtkPrintSettings* settings)
+        if (!gtk_check_version(4, 19, 3))
+            fileBackendName = "GtkPrintBackendFileBuiltin";
+        else
+            fileBackendName = "GtkPrintBackendFile";
```
```suggestion
        const char* fileBackendName = gtk_check_version(4, 19, 3) ? "GtkPrintBackendFile" : "GtkPrintBackendFileBuiltin" ;
```

carlosgcampos replied:
Is the condition correct? isn't GtkPrintBackendFileBuiltin the new name?

mcatanzaro replied:
The condition is correct, and GtkPrintBackendFileBuiltin is the new name.

The API is confusing: it returns NULL on success, or -- for some reason -- a string on failure.

I will change it to use the ternary operator, though.

mcatanzaro committed:
gtk_printer_get_backend: assertion 'GTK_IS_PRINTER (printer)' failed
https://bugs.webkit.org/show_bug.cgi?id=296238

Reviewed by Carlos Garcia Campos.

The GtkPrintBackend is now bundled into libgtk rather than installed
separately on the filesystem, and its name has changed.
WebKitPrintOperation needs to be updated accordingly, or printing does
not work.

Also, bail out if filePrinter is nullptr.

I tested this by manually hacking shouldUsePortal to return true.

Canonical link: https://commits.webkit.org/297692@main
